jwql/instrument_monitors/miri_monitors/cosmic_ray_monitor.py

```diff
@@ -0,0 +1,557 @@
+from jwql.database.database_interface import NIRCamCosmicRayStats
+from jwql.database.database_interface import session
+from jwql.instrument_monitors import pipeline_tools
+from jwql.jwql_monitors import monitor_mast
+from jwql.utils.constants import JWST_INSTRUMENT_NAMES, JWST_INSTRUMENT_NAMES_MIXEDCASE, JWST_DATAPRODUCTS
+from jwql.utils.logging_functions import configure_logging
+from jwql.utils.logging_functions import log_info
+from jwql.utils.logging_functions import log_fail
+from jwql.utils.utils import copy_files, ensure_dir_exists, get_config
+
+
+class CosmicRay:
+    """Class for executing the cosmic ray monitor.
+
+    This class will search for new (since the previous instance of the
+    class) MIRI and NIRCam data in the file system. It will loop over
+    instrument/aperture combinations and find the number of new files
+    available. It will copy the files over to a working directory and
+    run the monitor. This will count the number and magnitude of all
+    cosmic rays in each new exposure. Results are all saved to
+    database tables.
+
+    Attributes
+    ----------
+    output_dir : str
+        Path into which outputs will be placed
+
+    data_dir : str
+        Path into which new files will be copied to be worked on
+
+    query_start : float
+        MJD start date to use for querying MAST
+
+    query_end : float
+        MJD end date to use for querying MAST
+
+    instrument : str
+        Name of instrument used to collect the dark current data
+
+    aperture : str
+        Name of the aperture used for the dark current (e.g.
+        ``NRCA1_FULL``)
+
+    query_table : sqlalchemy table
+        Table containing the history of cosmic ray monitor queries to
+        MAST for each instrument/aperture combination
+
+    stats_table : sqlalchemy table
+        Table containing cosmic ray analysis results. Number and
+        magnitude of cosmic rays, etc.
+
+    Raises
+    ------
+    ValueError
+        If encountering a file not following the JWST file naming
+        convention
+
+    ValueError
+        If the most recent query search returns more than one entry
+    """
+
+    def __init__(self):
+        """Initialize an instance of the ``CosmicRay`` class."""
+
+    def get_cr_mags(self, jump_locations, jump_locations_pre, rateints, jump_data, jump_head):
+        """Gets the magnitude of each cosmic ray.
+
+        Computes a list of magnitudes using the coordinate of the
+        detected jump compared to the magnitude of the same pixel in
+        the group prior to the jump.
+
+        Parameters
+        ----------
+        jump_locations : list
+            List of coordinates to a pixel marked with a jump.
+
+        jump_locations_pre : list
+            List of matching coordinates one group before
+            ``jump_locations``.
+
+        rateints : ndarray obj
+            Array in ``DN/s``.
+
+        jump_data :
+
+        jump_head :
```

Comment:
These parameters were missing in the docstring, and I can't quite deduce what they are for based on the code itself.  Would you mind filling them in?